.github/workflows/publish-prerelease.yml

```diff
@@ -0,0 +1,60 @@
+name: Publish prerelease
+
+on:
+  workflow_call:
+    secrets:
+      PR_COMMENT_TOKEN:
+        required: true
+
+jobs:
+  publish-prerelease:
+    name: Publish prerelease
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        uses: metamask/github-tools/.github/actions/setup-environment@main
+
+      - name: Get CircleCI job details
+        id: get-circleci-job-details
+        env:
+          REPOSITORY: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq -r ".items[0].id")
```

Comment:
This will get the wrong pipeline in the case of two successive commits: both will get the "most recent" pipeline, which both correspond to the second commit. Mixing up runs initiated by different commits will make CI results confusing to interpret.

Could we filter the results here by revision? It looks like the revision is available as `item[*].vcs.revision`

Reply:
Latest workflow should work though I think. I tested it, and even when you only re-run certain jobs, the results from the jobs _not_ run are carried forward. So using the latest run at all times seems correct.

Reply:
I see the existing "wait for CircleCI" workflow uses a similar strategy. Here is my attempt at fixing that: https://github.com/MetaMask/metamask-extension/pull/29310
(not yet tested, and still needs a bit of cleanup)